functions/index.js

```diff
@@ -0,0 +1,153 @@
+const axios = require('axios');
+const btoa = require('btoa')
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+
+admin.initializeApp({
+    credential: admin.credential.applicationDefault(),
+    databaseURL: "https://tuneiq.firebaseio.com"
+});
+
+exports.createGame = functions.https.onRequest((request, response) => {
+    const gameGenre = request.query.genre
+    const hostId = request.query.hostId
+    const gameName = request.query.name
+    const password = request.query.password
+    const gameRounds = functions.config().gameconfig.number_of_rounds
+
+    getRoundDetails(gameGenre, gameRounds).then(roundDetails => {
+        gameDetails = {
+            hostId: hostId,
+            name: gameName,
+            password: password,
+            genre: gameGenre,
+            roundDetails: roundDetails,
+            players: [hostId]
```

Comment:
i think we should just have a leaderBoard object like { [playerId]: score } to keep track of users in the game

Reply:
Do we want to keep the leaderboard ordered whenever players' scores change or should the client just take care of ordering?

Reply:
talked today, the ordering will be done in the front end